Libraries/LibJS/Runtime/ProxyObject.h

```diff
@@ -26,19 +26,24 @@
 
 #pragma once
 
-#include <LibJS/Runtime/Object.h>
+#include <LibJS/Runtime/Function.h>
 
 namespace JS {
 
-class ProxyObject : public Object {
-    JS_OBJECT(ProxyObject, Object);
+class ProxyObject final : public Function {
+    JS_OBJECT(ProxyObject, Function);
 
 public:
     static ProxyObject* create(GlobalObject&, Object& target, Object& handler);
 
     ProxyObject(Object& target, Object& handler, Object& prototype);
     virtual ~ProxyObject() override;
 
+    Value call(Interpreter& interpreter) override;
+    Value construct(Interpreter& interpreter) override;
+    const FlyString& name() const override;
+    LexicalEnvironment* create_environment() override;
```

Comment:
Please mark these `virtual`
Also please omit the parameter names for `call` and `construct` since they are identical to the type name.